controllers/etcdpeer_controller_test.go

```diff
@@ -1,62 +1,52 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
 package controllers
 
 import (
-	"context"
+	"github.com/avast/retry-go"
+	"testing"
 	"time"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
 	appsv1 "k8s.io/api/apps/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	etcdv1alpha1 "github.com/improbable-eng/etcd-cluster-operator/api/v1alpha1"
 )
 
-var _ = Describe("Etcd peer controller", func() {
-	ctx := context.Background()
-	SetupTest(ctx)
-
-	Context("ReplicaSet creation", func() {
-		It("Should create a ReplicaSet if one does not exist already", func() {
-
-			By("Creating a EtcdPeer")
-			etcdPeer := etcdv1alpha1.EtcdPeer{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels:      make(map[string]string),
-					Annotations: make(map[string]string),
-					Name:        "bees",
-					Namespace:   "default",
-				},
-			}
-
-			err := k8sClient.Create(ctx, &etcdPeer)
-			Expect(err).NotTo(HaveOccurred(), "failed to create EtcdPeer resource")
-
-			By("The controller having created a ReplicaSet")
-			replicaSet := &appsv1.ReplicaSet{}
-			Eventually(
-				getResourceFunc(
-					ctx,
-					client.ObjectKey{
-						// Same name and namespace as the EtcdPeer above
-						Name:      etcdPeer.Name,
-						Namespace: etcdPeer.Namespace,
-					},
-					replicaSet,
-				),
-				// Timeout
-				time.Second*5,
-				// Polling Interval
-				time.Millisecond*500,
-			).Should(BeNil())
-		})
-	})
-})
+func (s *controllerSuite) testPeerController(t *testing.T) {
+	t.Run("TestPeerController_OnCreation_CreatesReplicaSet", func(t *testing.T) {
+		teardownFunc := s.setupTest(t)
+		defer teardownFunc()
+
+		etcdPeer := &etcdv1alpha1.EtcdPeer{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels:      make(map[string]string),
+				Annotations: make(map[string]string),
+				Name:        "bees",
+				Namespace:   "default",
+			},
+		}
+
+		err := s.k8sClient.Create(s.ctx, etcdPeer)
+		require.NoError(t, err, "failed to create EtcdPeer resource")
+
+		replicaSet := &appsv1.ReplicaSet{}
```

Comment:
This isn't a comment about your code, as I'm the one that wrote that bit, but as an aside is there a more idiomatic way of creating an empty pointer in go? Here we create a blank `appsv1.ReplicaSet` struct, only to throw it away when the `k8sClient.Get` function executes later.

Reply:
not afaik. i guess we could clean this up a little by defining it inline, as its never used outside of that scope

Reply:
One thing I realised I omitted from this patch was asserting things about the Replica Set. so we do need it at this level. It's no matter, hopefully the compiler is smart enough to realise it's never used and so might not bother to actually create it.

Reply:
You could define a non-ptr type and then point to it, e.g.

```golang
var replicaSet appsv1.ReplicaSet
doSomething(&replicaSet)
```

but what you've done here is also pretty idiomatic 😄